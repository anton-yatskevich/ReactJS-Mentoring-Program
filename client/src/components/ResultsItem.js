import React from 'react';
import PropTypes from 'prop-types';

const ResultsItem = ({
    poster_path: path, title, genres, id, release_date: date, onSelectHandler
}) => (
    <div className="results-item">
        <img className="results-item__poster" src={path} alt={title} role="presentation" onClick={() => onSelectHandler(id)} />
        <p className="results-item__description-header">
            <span className="results-item__title">{title}</span>
            <span className="results-item__date">{date.slice(0, 4)}</span>
        </p>
        <p>{genres.join(' & ')}</p>
    </div>
);

ResultsItem.propTypes = {
    poster_path: PropTypes.string,
    title: PropTypes.string,
    id: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    release_date: PropTypes.string,
    onSelectHandler: PropTypes.func
};

ResultsItem.defaultProps = {
    poster_path: '',
    title: '',
    genres: [],
    id: 0,
    release_date: '',
    onSelectHandler: null
};

export default ResultsItem;
