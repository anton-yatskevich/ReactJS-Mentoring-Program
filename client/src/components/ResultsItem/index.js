import React from 'react';
import { withRouter } from 'next/router';
import styled from 'styled-components';
import MovieSchema from '../../constants/MovieSchema';
import { Router } from '../../routes';

const ResultsItem = ({ movie }) => {
    const {
        poster_path: path, title, genres, release_date: date
    } = movie;

    return (
        <Result>
            <Poster src={path} alt={title} role="presentation" onClick={() => Router.pushRoute(`/film/${movie.id}`)} />
            <Description>
                <Title>{title}</Title>
                <span>{date && date.slice(0, 4)}</span>
            </Description>
            <p>{genres && genres.join(' & ')}</p>
        </Result>
    );
};

ResultsItem.propTypes = {
    movie: MovieSchema
};

ResultsItem.defaultProps = {
    movie: {}
};

const Result = styled.div`
    margin: 10px;
    max-width: 280px;
`;

const Poster = styled.img`
    width: 100%;
    margin-bottom: 5px;
    cursor: pointer;
`;

const Description = styled.p`
    display: flex;
    justify-content: space-between;
    margin-bottom: 5px;
`;

const Title = styled.span`
    font-weight: bold;
`;

export default withRouter(ResultsItem);
