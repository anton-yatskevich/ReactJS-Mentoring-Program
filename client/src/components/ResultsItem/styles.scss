.results-item {
	margin: 10px;
    max-width: 280px;

	&__poster {
		width: 100%;
		margin-bottom: 5px;
		cursor: pointer;
	}

	&__description-header {
		display: flex;
		justify-content: space-between;
		margin-bottom: 5px;
	}

	&__title {
		font-weight: bold;
	}
}