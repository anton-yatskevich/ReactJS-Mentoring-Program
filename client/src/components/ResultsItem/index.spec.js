import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ResultsItem from './index';

describe('ResultsItem component', () => {
    it('should be rendered with default props', () => {
        const ResultsItemComponent = shallow(<ResultsItem />);
        expect(toJson(ResultsItemComponent)).toMatchSnapshot();
    });
});
