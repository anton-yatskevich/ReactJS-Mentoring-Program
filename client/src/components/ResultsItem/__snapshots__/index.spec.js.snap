// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsItem component should be rendered with default props 1`] = `
<div
  className="results-item"
>
  <img
    alt=""
    className="results-item__poster"
    onClick={[Function]}
    role="presentation"
    src=""
  />
  <p
    className="results-item__description-header"
  >
    <span
      className="results-item__title"
    />
    <span
      className="results-item__date"
    />
  </p>
  <p />
</div>
`;
